### DISCUSSION TITLE
Can Ascii GameCube Keyboard Controller Support be Added

### DISCUSSION THREAD

**Discussion by Wii505** [KeyPuncher/WiinUPro#159] [Ideas]
I just saw Michael MJD's video about the Ascii GameCube Keyboard Controller and he said that you have to get a specific adapter to make the keyboard part work on Windows while using their software. He also said that it don't work with Nintendo's official adapter on windows. And I thought that using the controller on windows would be cool, but I don't want to buy another adapter in order to make the keyboard part work. So, if support is added for the controller is add would be huge and appreciated

**Comment by KeyPuncher** [comment]
That would be really cool. I found [this page](https://blarg.ca/2021/02/15/using-a-gamecube-ascii-keyboard-controller-with-libogc/) from over 2 years ago about someone looking into how it works. They also have a [GitHub Repo](https://github.com/gered/gc-ascii-keyboard/tree/master) with their findings.

Since I don't have one, in order to get it implemented I'll have to have create some sort of debugging tools. I need that for one of the other issues too.

**Comment by KeyPuncher** [comment]
[Debug.zip](https://github.com/KeyPuncher/WiinUPro/files/12903514/Debug.zip)
This build has a debugger in it. Right click on the Connect button to get to the Debug option before connecting.
You should get a side window open. Clicking the capture button will grab the data I need to look at for the current state.

For your keyboard controller you're going to have to make a lot of notes.
The first capture should be when it is just plugged in and nothing is pressed.
To verify the data is reading, try holding down the GCN's A button when doing the second capture, and make sure you see one of the characters change in the text.
Next you'll need to painstakingly do a capture while holding down each individual key to see what the data reports if anything at all.
